### PR TITLE
Add GITHUB_ALWAYS_PASS configuration option

### PR DESCRIPTION
**Pull request by zackgalbreath**
When this environment variable is set to true, CDash will always report a status of 'success' to any GitHub repositories that have installed this CDash server as a GitHub App.

This is useful in cases where you do not want the CDash GitHub check to prevent a PR from being merged.

### CODE REVIEW

**Comment by zackgalbreath** [comment]
You can see an example of this new configuration option in action here:

https://github.com/zackgalbreath/TestProject/pull/1/checks

The GitHub Actions test fails (on purpose), but the CDash check is still ✅

File: app/cdash/app/Lib/Repository/GitHub.php [hunk 1]
```diff
@@ -89,19 +82,22 @@ public function __construct(Project $project)
...
     {
         $builder = new GitHubBuilder();
         $apiClient = new GithubClient($builder, 'machine-man-preview');
         $this->setApiClient($apiClient);
     }
 
-    public function authenticate($required = true)
+    /**
+      * Attempt to log in to the GitHub API
+      */
```
Comment (williamjallen) [inline app/cdash/app/Lib/Repository/GitHub.php:99]:
```suggestion
    /**
     * Attempt to log in to the GitHub API
     */
```

File: app/cdash/app/Lib/Repository/GitHub.php [hunk 2]
```diff
@@ -384,17 +379,25 @@ public function generateCheckPayloadFromBuildRows($build_rows, $head_sha)
...
+
         return $params;
     }
 
     /**
      * Generate a check summary for a given row of build data.
+     * @param array<string, int|string> $row
      */
-    public function getCheckSummaryForBuildRow($row)
+    public function getCheckSummaryForBuildRow($row) : string|null
```
Comment (williamjallen) [inline app/cdash/app/Lib/Repository/GitHub.php:396]:
```suggestion
    public function getCheckSummaryForBuildRow(array $row) : string|null
```

**Review by williamjallen — COMMENTED** [review]
I don't have a way to test it easily, but the code looks reasonable in general.

**Pull request merged**